blockchain/indexers/indexsubscriber.go

```diff
@@ -80,6 +81,8 @@ func newIndexSubscription(subber *IndexSubscriber, indexer Indexer, prereq strin
 
 // stop prevents any future index updates from being delivered and
 // unsubscribes the associated subscription.
+//
+// This must be called with the index subscriber mutex held for writes.
```

Comment:
What is "the index subscriber mutex"?.  If it's `s.mtx`, you'll have a deadlock in the `if s.prerequisite != noPrereqs {` branch.

Reply:
`s.mtx` here would be the index subscription mutex, the index subscriber mutex from, accessing from the subscription would be `s.subscriber.mtx`.